project/src/components/favorites-location/favorites-location.tsx

```diff
@@ -1,22 +1,55 @@
 import CardList from '../card-list/card-list';
+import {Offers} from '../../types/offers';
 
-const cards = Array.from({length: 2}, (_, i) => i);
+type FavoritesLocationProps = {
+  offers: Offers;
+}
+
+type CityAndOffers = {
+  city: string;
+  cityOffers: Offers;
+}
+
+const spliceOffersByCity = (offers: Offers): Map<string, Offers> => {
+  const sortedOffers = new Map();
+  let prevCity = null;
+
+  for(const offer of offers) {
+    const city = offer.city.name;
+    if (prevCity === city) {
+      sortedOffers.get(city).push(offer);
+    } else {
+      sortedOffers.set(city, [offer]);
+    }
+    prevCity = city;
+  }
 
-function FavoritesLocation(): JSX.Element {
+  return sortedOffers;
+};
+
+function FavoritesLocationsByCity ({city, cityOffers}: CityAndOffers): JSX.Element {
   return (
     <li className='favorites__locations-items'>
       <div className='favorites__locations locations locations--current'>
         <div className='locations__item'>
           <a className='locations__item-link' href='/#'>
-            <span>Amsterdam</span>
+            <span>{city}</span>
           </a>
         </div>
       </div>
       <div className='favorites__places'>
-        <CardList cards={cards} cardType={'favorites'}/>
+        <CardList offers={cityOffers} cardType={'favorites'} />
       </div>
     </li>
   );
 }
 
-export default FavoritesLocation;
+function FavoritesLocations({offers}: FavoritesLocationProps): JSX.Element {
+  return (
+    <div>{Array.from(spliceOffersByCity(offers), ([city, cityOffers]) => ({city, cityOffers}))
+      .map((cityAndOffers) => FavoritesLocationsByCity(cityAndOffers))}
```

Comment:
Почему ты компонент как функцию вызываешь? Надо передавать пропсы по отдельности 
```suggestion
      .map((cityAndOffers) => <FavoritesLocationsByCity city={city} offers={offers}/>))}
```